in_toto/models/metadata.py

```diff
@@ -23,20 +23,195 @@
+        return Envelope.from_dict(data)
+
+    elif "signed" in data:
+      return Metablock.from_dict(data)
+
+    raise InvalidMetadata
+
+  def to_dict(self) -> dict:
+    """Abstractmethod to convert metadata into dict"""
+    raise NotImplementedError  # pragma: no cover
+
+  @classmethod
+  def load(cls, path) -> Union["Envelope", "Metablock"]:
+    """Loads the JSON string representation of metadata from disk.
+
+    Arguments:
+      path: The path to read the file from.
+
+    Raises:
+      IOError: The file cannot be read.
+      securesystemslib.exceptions.FormatError: Metadata format is invalid.
+
+    Returns:
+      A Metablock object whose signable attribute is either a Link or a Layout
+      object.
+
+    """
+    with open(path, "r", encoding="utf8") as fp:
+      data = json.load(fp)
+
+    return cls.from_dict(data)
+
+  def dump(self, path, compact=False):
+    """Writes the JSON string representation of the instance to disk.
+
+    Arguments:
+      path: The path to write the file to.
+
+    Raises:
+      IOError: File cannot be written.
+
+    """
+    indent = None if compact else 1
+    separators = (',', ':') if compact else (',', ': ')
+
+    json_bytes = json.dumps(
+      self.to_dict(),
+      indent=indent,
+      separators=separators,
+      sort_keys=True,
+    ).encode("utf-8")
+
+    with open(path, "wb") as fp:
+      fp.write(json_bytes)
+
+  def create_signature(self, signer: Signer) -> Signature:
+    """Creates signature over signable with Signer and adds it to signatures.
+
+    Uses the UTF-8 encoded canonical JSON byte representation of the signable
+    attribute to create signatures deterministically.
+
+    Arguments:
+      signer: A "Signer" class instance to create signature.
+
+    Raises:
+      securesystemslib.exceptions.FormatError: Key argument is malformed.
+      securesystemslib.exceptions.CryptoError, \
+              securesystemslib.exceptions.UnsupportedAlgorithmError:
+          Signing errors.
+
+    Returns:
+      The signature. A securesystemslib.signer.Signature type.
+    """
+    raise NotImplementedError  # pragma: no cover
+
+  def verify_signatures(
+    self,
+    keys: List[Key],
+    threshold: int
+  ) -> Dict[str, Key]:
+    """Verifies signature over signable in signatures with a list of keys.
+
+    Uses the UTF-8 encoded canonical JSON byte representation of the signable
+    attribute to verify the signature deterministically.
+
+    Arguments:
+        keys: A list of public keys to verify the signatures.
+        threshold: Number of signatures needed to pass the verification.
+
+    Raises:
+        ValueError: If "threshold" is not valid.
+        VerificationError: If the enclosed signatures do not pass the
+          verification.
+
+    Returns:
+        accepted_keys: A dict of unique public keys.
+    """
+    raise NotImplementedError  # pragma: no cover
+
+  def get_payload(self):
+    """Returns ``Link`` or ``Layout`` from the metadata wrapper."""
```

Comment:
Minor nit: We don't use the term "metadata wrapper" anywhere else. Might be enough to just say
```suggestion
    """Returns ``Link`` or ``Layout``."""
```